crates/bevy_ui/src/focus.rs

```diff
@@ -175,16 +196,31 @@ pub fn ui_focus_system(
                 let ui_position = position.truncate();
                 let extents = node.node.size() / 2.0;
                 let mut min = ui_position - extents;
-                let mut max = ui_position + extents;
                 if let Some(clip) = node.calculated_clip {
                     min = Vec2::max(min, clip.clip.min);
-                    max = Vec2::min(max, clip.clip.max);
                 }
-                // if the current cursor position is within the bounds of the node, consider it for
+
+                // The mouse position relative to the node
+                // (0., 0.) is the upper-left corner, (1., 1.) is the bottom-right corner
+                let relative_cursor_postition = cursor_position.map(|cursor_position| {
+                    Vec2::new(
+                        (cursor_position.x - min.x) / node.node.size().x,
+                        (cursor_position.y - min.y) / node.node.size().y,
+                    )
+                });
+
+                // If the current cursor position is within the bounds of the node, consider it for
                 // clicking
-                let contains_cursor = if let Some(cursor_position) = cursor_position {
-                    (min.x..max.x).contains(&cursor_position.x)
-                        && (min.y..max.y).contains(&cursor_position.y)
+                let contains_cursor = if let Some(cursor_position) = relative_cursor_postition {
+                    // Save the relative cursor position to the correct component
+                    if let Some(mut relative_cursor_position_component) =
+                        node.relative_cursor_position
+                    {
+                        relative_cursor_position_component.0 = cursor_position;
+                    }
+
+                    ((0.)..1.).contains(&cursor_position.x)
+                        && ((0.)..1.).contains(&cursor_position.y)
```

Comment:
That `(0.)` feels really weird, why not just use `0.0` instead?
```suggestion
                    (0.0..1.).contains(&cursor_position.x)
                        && (0.0..1.).contains(&cursor_position.y)
```